builtin/providers/aws/resource_aws_route53_health_check.go

```diff
@@ -0,0 +1,226 @@
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"fully_qualified_domain_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"port": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+
+			"resource_path": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"search_string": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsRoute53HealthCheckUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).r53conn
+
+	updateHealthCheck := &route53.UpdateHealthCheckInput{
+		HealthCheckID: aws.String(d.Id()),
+	}
+
+	if d.HasChange("failure_threshold") {
+		updateHealthCheck.FailureThreshold = aws.Long(int64(d.Get("failure_threshold").(int)))
+	}
+
+	if d.HasChange("fully_qualified_domain_name") {
+		updateHealthCheck.FullyQualifiedDomainName = aws.String(d.Get("fully_qualified_domain_name").(string))
+	}
+
+	if d.HasChange("port") {
+		updateHealthCheck.Port = aws.Long(int64(d.Get("port").(int)))
+	}
+
+	if d.HasChange("resource_path") {
+		updateHealthCheck.ResourcePath = aws.String(d.Get("resource_path").(string))
+	}
+
+	if d.HasChange("search_string") {
+		updateHealthCheck.SearchString = aws.String(d.Get("search_string").(string))
+	}
+
+	_, err := conn.UpdateHealthCheck(updateHealthCheck)
+	if err != nil {
+		return err
+	}
+
+	if err := setTagsR53(conn, d, "healthcheck"); err != nil {
+		return err
+	}
+
+	return resourceAwsRoute53HealthCheckRead(d, meta)
+}
+
+func resourceAwsRoute53HealthCheckCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).r53conn
+
+	// do we need to check if the optional fields existf before adding them?
+	healthConfig := &route53.HealthCheckConfig{
+		Type:             aws.String(d.Get("type").(string)),
+		FailureThreshold: aws.Long(int64(d.Get("failure_threshold").(int))),
+		RequestInterval:  aws.Long(int64(d.Get("request_interval").(int))),
+	}
+
+	if v, ok := d.GetOk("fully_qualified_domain_name"); ok {
+		healthConfig.FullyQualifiedDomainName = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("search_string"); ok {
+		healthConfig.SearchString = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("ip_address"); ok {
+		healthConfig.IPAddress = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("port"); ok {
+		healthConfig.Port = aws.Long(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("resource_path"); ok {
+		healthConfig.ResourcePath = aws.String(v.(string))
+	}
+
+	input := &route53.CreateHealthCheckInput{
+		CallerReference:   aws.String(time.Now().Format(time.RFC3339Nano)),
+		HealthCheckConfig: healthConfig,
+	}
+
+	wait := resource.StateChangeConf{
+		Pending:    []string{"rejected"},
+		Target:     "accepted",
+		Timeout:    5 * time.Minute,
+		MinTimeout: 1 * time.Second,
+		Refresh: func() (interface{}, string, error) {
```

Comment:
Does this process typically take some time? If so, we're not really waiting on anything here, we just immediately reply `"accepted"` if there's no error, regardless of what the state actually is. 

If it's a quick operation that either succeeds or fails, there's no need for a refresh func. If it can take time, you should correctly populate the states here and let the function retry appropriately. At which point, you're likely wanting to call `CreateHealthCheck` separately, and maybe poll for it's status. 

Does that make sense?